ui/src/core_components/embed/CorePDF.vue

```diff
@@ -33,26 +60,251 @@ export default {
 				type: FieldType.Text,
 				desc: "A valid URL. Alternatively, you can provide a state reference to a packed PDF file.",
 			},
+			highlights: {
+				name: "Highlights",
+				default: JSON.stringify([]),
+				desc: "A list of highlights to be applied to the PDF as a JSON array of strings.",
+				type: FieldType.Object,
+			},
+			selectedMatch: {
+				name: "Selected highlight match",
+				default: null,
+				desc: "The index of the selected highlight match.",
+				type: FieldType.Number,
+			},
+			page: {
+				name: "Page",
+				type: FieldType.Number,
+				desc: "The page to be displayed.",
+			},
 			cssClasses,
 		},
+		events: {
+			"pdf-load": {
+				desc: "",
+			},
+		},
 	},
 };
 </script>
 
 <script setup lang="ts">
-import { inject } from "vue";
+import { inject, ref, watch, computed } from "vue";
+import { VuePDF, usePDF } from "@tato30/vue-pdf";
 import injectionKeys from "../../injectionKeys";
+import "@tato30/vue-pdf/style.css";
+
+type MatchType = { str: string; page: number; index: number };
+
 const fields = inject(injectionKeys.evaluatedFields);
+
+const { pdf, pages } = usePDF(fields.source);
+
+const highlightText = computed(() => {
+	return fields.highlights.value
+		? fields.highlights.value
+				.map((v: string) => v.replaceAll("|", "\\|"))
+				.join("|")
+		: "";
+});
+const highlightOptions = ref({
+	completeWords: false,
+	ignoreCase: true,
+});
+const scale = ref(1);
+const page = ref(1);
+const matches = ref<MatchType[]>([]);
+const currentMatch = ref(0);
+const rootEl = ref(null);
+const viewerEl = ref(null);
+
+const onHighlight = (value) => {
+	matches.value = [
+		...matches.value,
+		...value.matches.map((m: MatchType, idx: number) => ({
+			page: value.page,
+			str: m.str,
+			index: idx,
+		})),
+	];
+	// sort by page and index
+	matches.value.sort((a, b) => {
+		if (a.page === b.page) {
+			return a.index - b.index;
+		}
+		return a.page - b.page;
+	});
+	if (currentMatch.value > 0) {
+		gotoHighlight(currentMatch.value);
+	}
+};
+
+const scroll = (event) => {
+	const c = event.target.getBoundingClientRect();
+	const r = [...event.target.children]
+		.filter((child) => child.className.includes("page"))
+		.find((child) => {
+			const e = child.getBoundingClientRect();
+			const relativeY = e.y - c.y;
+			return (
+				relativeY < c.height / 2 && relativeY + e.height > c.height / 2
+			);
+		});
+	if (r) {
+		page.value = Number(r.getAttribute("page"));
+	}
+};
+
+const calcScrollPosition = (targetEl: HTMLElement, parentEl: HTMLElement) => {
+	let offsetTop = 0;
+	let el = targetEl;
+	while (el && el !== parentEl) {
+		offsetTop += el.offsetTop;
+		el = el.offsetParent as HTMLElement;
+	}
+	return offsetTop;
+};
+
+const gotoHighlight = (matchIdx: number) => {
+	if (matchIdx < 1 || matchIdx > matches.value.length) {
+		return;
+	}
+	const match = matches.value[matchIdx - 1];
+	console.log(match);
```

Comment:
Please remove

Reply:
Also, I once remember reading about some guy having a linting rule to get rid of `console.log` messages. Since you're good at that stuff maybe you can implement something similar?

I'd still want to keep the loading messages (legitimate uses of `console.log`) but that should probably be handled in a more professional way, defining a logger and then invoking that logger.

Reply:
👍 
Rule added to eslint
you can use:
```js
// eslint-disable-next-line no-console
```
if you need to keep logging

Reply:
Great stuff